tt_metal/distributed/mesh_device.hpp

```diff
@@ -117,9 +124,11 @@ class MeshDevice : public std::enable_shared_from_this<MeshDevice> {
     size_t num_cols() const;
     MeshShape shape() const;
 
+    // Reshape changes the shape of the logical mesh and re-maps the physical devices to the new logical coordinates.
```

Comment:
`Reshapes`. Also, is it easy to outline all of the conditions when it is possible in a comment here?

Reply:
Updated to describe the rule-set. Thanks!